app/src/main/java/ml/docilealligator/infinityforreddit/markdown/SuperscriptPlugin.java

```diff
@@ -0,0 +1,200 @@
+            int spanFlags = spannableStringBuilder.getSpanFlags(span);
+            spanList.add(new SpanInfo(span, spanStart, spanEnd, spanFlags));
+        }
+        return spanList;
+    }
+
+    private static SpanInfo matchSpanAtPosition(List<SpanInfo> spans, int value, Object spanClass) {
+        for (var span : spans)
+            if (span.what.getClass() == spanClass && span.start <= value && value <= span.end)
+                return span;
+        return null;
+    }
+
+    private static SpanInfo matchNonTextSpanAtBoundary(List<SpanInfo> spans, int value) {
+        for (var span : spans)
+            if (span.what.getClass() != TextViewSpan.class && (span.end == value || span.start == value))
+                return span;
+        return null;
+    }
+
+    @Override
+    public void configureSpansFactory(@NonNull MarkwonSpansFactory.Builder builder) {
+        builder.setFactory(Superscript.class, (config, renderProps) -> new SuperscriptSpan(true));
+    }
+
+    @Override
+    public void configureVisitor(@NonNull MarkwonVisitor.Builder builder) {
+        builder.on(Superscript.class, new MarkwonVisitor.NodeVisitor<>() {
+            int depth = 0;
+
+            @Override
+            public void visit(@NonNull MarkwonVisitor visitor, @NonNull Superscript superscript) {
+                int start = visitor.length();
+
+                if (!superscript.isBracketed()) {
+                    visitor.builder().setSpan(new SuperscriptSpan(false), start, start + 1); // Workaround for Table Plugin
+                    superscriptOpeningList.add(new SuperscriptOpening(superscript, start));
+                    return;
+                }
+
+                depth++;
+                visitor.visitChildren(superscript);
+                depth--;
+                if (depth == 0) {
+                    int end = visitor.builder().length();
+                    var spannableStringBuilder = visitor.builder().spannableStringBuilder();
+                    var codeSpans = spannableStringBuilder.getSpans(start, end, CodeSpan.class);
+                    if (codeSpans.length > 0) {
+                        for (int i = codeSpans.length - 1; i >= 0; i--) {
+                            var span = codeSpans[i];
+                            var spanStart = spannableStringBuilder.getSpanStart(span);
+                            var spanEnd = spannableStringBuilder.getSpanEnd(span);
+                            var nextSpan = i - 1 >= 0 ? codeSpans[i - 1] : null;
+                            var nextSpanStart = nextSpan == null ? null : spannableStringBuilder.getSpanStart(nextSpan);
+                            if (spanEnd <= end) {
+                                visitor.builder().setSpan(new SuperscriptSpan(true), start, spanStart);
+                                visitor.builder().setSpan(new SuperscriptSpan(true), spanEnd, nextSpanStart == null ? end : nextSpanStart);
+                                start = spanEnd;
+                            }
```

Comment:
`d` in the middle is still spanned twice
```suggestion
                            if (start < spanStart) {
                                visitor.builder().setSpan(new SuperscriptSpan(true), start, spanStart);
                            }
                            if (nextSpan == null && spanEnd < end) {
                                visitor.builder().setSpan(new SuperscriptSpan(true), spanEnd, end);
                            }
                            start = spanEnd;
```